pytest_mongo/factories/process.py

```
@@ -41,6 +48,34 @@ def mongo_proc_fixture(
+
+        n = 0
+        used_ports: set[int] = set()
+        while True:
+            try:
+                pg_port = _mongo_port(port, config, used_ports)
+                port_filename_path = port_path / f"mongo-{pg_port}.port"
+                if pg_port in used_ports:
+                    raise PortForException(
+                        f"Port {pg_port} already in use, probably by other instances of the test. "
+                        f"{port_filename_path} is already used."
+                    )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)